app/src/main/java/com/openclassrooms/realestatemanager/ui/EstateListAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package com.openclassrooms.realestatemanager.ui
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import com.openclassrooms.realestatemanager.R
+import com.openclassrooms.realestatemanager.model.Estate
+import java.text.DecimalFormat
+
+class EstateListAdapter(diffCallback: DiffUtil.ItemCallback<Estate>) : ListAdapter<Estate, EstateListAdapter.ViewHolder>(diffCallback) {
+
+    private val decimalFormat: DecimalFormat = DecimalFormat("$#,###.00")
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.item_estate, parent, false)
+        return ViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val estate: Estate = currentList[position]
```

Comment:
```suggestion
        val estate = currentList[position]
```

Le typage à l'initialisation est optionnel en Kotlin quand le type peut être déduit par la valeur assignée. Si tes variables sont bien nommées, il est préférable de ne pas expliciter pour alléger le code

Reply:
Ok